src/main/extension/extension_load.cpp

```diff
@@ -119,15 +120,50 @@ bool ExtensionHelper::TryInitialLoad(DBConfig &config, FileSystem &fs, const str
 		error = StringUtil::Format("Extension \"%s\" not found.\n%s", filename, message);
 		return false;
 	}
-	if (!config.options.allow_unsigned_extensions) {
-		auto handle = fs.OpenFile(filename, FileFlags::FILE_FLAGS_READ);
 
-		// signature is the last 256 bytes of the file
+	string metadata_segment;
```

Comment:
I think the `ExtensionHelper::TryInitialLoad` has reached a point here where it is a bit too long to be readable, could we split this one up in some smaller functions?